mongodbatlas/resource_mongodbatlas_online_archive.go

```diff
@@ -513,7 +513,6 @@ func fromOnlineArchiveToMap(in *matlas.OnlineArchive) map[string]interface{} {
 
 func fromOnlineArchiveToMapInCreate(in *matlas.OnlineArchive) map[string]interface{} {
```

Comment:
@marcosuma Since this method is also used in the read handler, can we test the plan output from upgrading to make sure there no unexpected/breaking changes to user configs who might be using this respurce?

Basically we need to:
 1. create this resource with a published version of our provider (can be 1.12.1)
 2. change to your local provider and run `terraform plan`
 3. confirm plan is empty

You can also automate the same by adding migration tests as in [https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1415/files#diff-fd02b9e0e35b68396ccd10d1300a8365a979efae778f5fe09a2ff1bbf30d0b5d](https://www.google.com/url?q=https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1415/files%23diff-fd02b9e0e35b68396ccd10d1300a8365a979efae778f5fe09a2ff1bbf30d0b5d&sa=D&source=docs&ust=1695901108249657&usg=AOvVaw0CrXLDWpruYzwKDe3CTOAB). For this case I'm okay with manual testing since it's one field only.

Reply:
Hi @maastha, this is the manual test I executed.

Had the same file defined above but with the only difference that `versions.tf` explicitly contained the version `1.12.1` and my debug locally was shut down.
So I ran `terraform apply` and successfully created online archive:
```
INTMDB-1136 terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # mongodbatlas_online_archive.this will be created
  + resource "mongodbatlas_online_archive" "this" {
      + archive_id      = (known after apply)
      + cluster_name    = "Cluster0"
      + coll_name       = "collectiontest"
      + collection_type = (known after apply)
      + db_name         = "databasetest"
      + id              = (known after apply)
      + paused          = (known after apply)
      + project_id      = "6513cd8f1a0e213da8cf71e2"
      + state           = (known after apply)
      + sync_creation   = false

      + criteria {
          + date_field        = "created"
          + date_format       = "ISODATE"
          + expire_after_days = 2
          + type              = "DATE"
        }

      + partition_fields {
          + field_name = "field_1"
          + field_type = (known after apply)
          + order      = 0
        }
      + partition_fields {
          + field_name = "created"
          + field_type = (known after apply)
          + order      = 1
        }

      + schedule {
          + end_hour     = 11
          + end_minute   = 0
          + start_hour   = 8
          + start_minute = 0
          + type         = "DAILY"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

mongodbatlas_online_archive.this: Creating...
mongodbatlas_online_archive.this: Creation complete after 3s [id=YXJjaGl2ZV9pZA==:NjUxNjZmYzllYzMxODMyYzhlMDg4OGIx-Y2x1c3Rlcl9uYW1l:Q2x1c3RlcjA=-cHJvamVjdF9pZA==:NjUxM2NkOGYxYTBlMjEzZGE4Y2Y3MWUy]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

Then I removed the version from the `versions.tf` file:
```
  1 terraform {                                                                                                                                    
  2   required_providers {
  3     mongodbatlas = {
  4       source  = "mongodb/mongodbatlas"
  5     }
  6   }
  7   required_version = ">= 1.0"
  8 }
```

Attached the debug from local and executed `terraform plan`
```
➜  INTMDB-1136 export TF_REATTACH_PROVIDERS='{"registry.terraform.io/mongodb/mongodbatlas":{"Protocol":"grpc","ProtocolVersion":6,"Pid":32600,"Test":true,"Addr":{"Network":"unix","String":"/var/folders/d5/77ngxh5d6r337g31ddhzfmfm0000gp/T/plugin1631226599"}}}'
➜  INTMDB-1136 terraform plan 
mongodbatlas_online_archive.this: Refreshing state... [id=YXJjaGl2ZV9pZA==:NjUxNjZmYzllYzMxODMyYzhlMDg4OGIx-Y2x1c3Rlcl9uYW1l:Q2x1c3RlcjA=-cHJvamVjdF9pZA==:NjUxM2NkOGYxYTBlMjEzZGE4Y2Y3MWUy]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
```

Now I am also trying to add a migration test, but not sure it's the best idea. I think we should find a good way to run migration tests at the moment of a new release that are always checking version-{-1} VS version-{0}

Reply:
>Now I am also trying to add a migration test, but not sure it's the best idea.

I am not completely sure I understand your concern here. FYI we are having this conversation in the test improvements proposal, so please feel free to share you r concerns there, I will share link in Slack

Reply:
@maastha  not major concern but I don't know if we should have "backward-compatibility-check" tests within our acceptance tests. It's not intuitive and easy to understand and it adds up yet another test in our suite which is already slow.

I am in favour as I mentioned of having migration tests but to be done separately in different folders and ran only _ad-hoc_

Reply:
Hi Marco, 
Responding here as well for visibility:

yes that's the plan to have those tests run ad-hoc/on-demand if the developer or the reviewer want to run those for a check, We don't expect to always run those test as you said that would make our suite even slower.

Also, note that these tests would anyway be required to add for migration to Plugin Framework so it's not extra effort as well and IMO they are needed for Atlas SDK migration as well. We just plan to leverage those for ourselves to maintain backward compatibility.